docs/documentation/quartz-3.x/tutorial/using-quartz.md

```diff
@@ -12,70 +12,83 @@ in the paused state.
 
 Here's a quick snippet of code, that instantiates and starts a scheduler, and schedules a job for execution:
 
-__Using Quartz.NET__
+### Install Quartz.NET NuGets
+
+```sh
+Install-Package Microsoft.Extensions.Hosting
+Install-Package Quartz
+Install-Packake Quartz.Extensions.DependencyInjection
+Install-Packake Quartz.Extensions.Hosting
+Install-Packake Quartz.Serialization.Json
+```
+
+### Configure `Program.cs`
+
+A minimal example of configuring Quartz.NET with the Microsoft Hosting framework
+looks like this.
+
+```csharp
+using Microsoft.Extensions.Hosting;
+using Quartz;
+
+var builder = Host.CreateDefaultBuilder()
+    .ConfigureServices((cxt, services) =>
+    {
+        services.AddQuartz(q =>
+        {
+            q.UseMicrosoftDependencyInjectionJobFactory();
+        });
+        services.AddQuartzHostedService(opt =>
+        {
+            opt.WaitForJobsToComplete = true;
+        });
+    }).Build();
+
+await builder.RunAsync();
+```
+
+Let's add a job to this.
+
 
 ```csharp
-// construct a scheduler factory using defaults
-StdSchedulerFactory factory = new StdSchedulerFactory();
 
-// get a scheduler
-IScheduler scheduler = await factory.GetScheduler();
-await scheduler.Start();
+using Microsoft.Extensions.Hosting;
+using Quartz;
+
+var builder = Host.CreateDefaultBuilder()
+    .ConfigureServices((cxt, services) =>
+    {
+        services.AddQuartz(q =>
+        {
+            q.UseMicrosoftDependencyInjectionJobFactory();
+        });
+        services.AddQuartzHostedService(opt =>
+        {
+            opt.WaitForJobsToComplete = true;
+        });
+    }).Build();
+
+var schedulerFactory = builder.Services.GetRequiredService<ISchedulerFactory>();
+var scheduler = await schedulerFactory.GetScheduler();
 
 // define the job and tie it to our HelloJob class
-IJobDetail job = JobBuilder.Create<HelloJob>()
+var job = JobBuilder.Create<HelloJob>()
```

Comment:
Happy to switch back. My mental approach was that "knowing" it was `IJobDetail` didn't really help tell the story, of how to get started. It was one more thing that my brain had to process. Additionally, since there are no alternative interfaces that we can compare and contrast `IJobDetail` too - it again felt like an extra detail not helping me understand the very beginning of my Quartz journey. 

Thoughts? Again happy to switch back.

Reply:
I likea your reasoning, `var` should be fine. I guess interfaces will help with testing, hopefully. 

Many of the things are inherited from Java to make porting and syncing between releases easier. Java version hasn't really been updated in years, except their stale bot running rampant closing issues (and some very valid ones too). This is why Quartz 4 (the Duke Nukem Project) will have less worries in that regard and I'm open to change internal code flows and naming when it's more practical.

Reply:
Well snap. I'm looking forward to v4 then. Def feeling the `Java` vibe and looking forward to making quartz a streamlined silky library of awesomeness.

Reply:
And hey don't hesitate to suggest API changes via PRs! Some persisted models like configuration, keys, serialization format I'm more hesitant to change but anything that makes the experience more streamlined like you mentioned is always appreciated. `main` branch is the playground to go wild, there's separate `3.x` maintenance branch to go slow.

The docs come from main so need to remember sync to 4.x folder too at some point, but probably later when iterations have been done.